<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1496</width>
    <height>814</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="OriginalImage">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>30</y>
      <width>261</width>
      <height>161</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="text">
     <string>Original Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="GrayImage">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>210</y>
      <width>261</width>
      <height>161</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="text">
     <string>Gray Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OpenImageButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>390</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Open Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ClearProcessingButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>430</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear Processing</string>
    </property>
   </widget>
   <widget class="QPushButton" name="GrayButton">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>390</y>
      <width>101</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Gray</string>
    </property>
   </widget>
   <widget class="QPushButton" name="EqhistButton">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>430</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Eqhist</string>
    </property>
   </widget>
   <widget class="QPushButton" name="CloseAppButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>470</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Close App</string>
    </property>
   </widget>
   <widget class="QLabel" name="DenoiseImage">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>30</y>
      <width>261</width>
      <height>161</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="text">
     <string>Denoise Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="ThresholdImage">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>210</y>
      <width>261</width>
      <height>161</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="text">
     <string>Threshold Image</string>
    </property>
   </widget>
   <widget class="QLabel" name="MorphologyImage">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>30</y>
      <width>261</width>
      <height>161</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="text">
     <string>Morphology Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="BlurButton">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>390</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Blur</string>
    </property>
   </widget>
   <widget class="QPushButton" name="MedianBlurButton">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>430</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Median Blur</string>
    </property>
   </widget>
   <widget class="QPushButton" name="GaussianBlurButton">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>470</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Gaussian Blur</string>
    </property>
   </widget>
   <widget class="QPushButton" name="BilateralBlurButton">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>510</y>
      <width>121</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Bilateral Blur</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ThresholdButton">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>390</y>
      <width>81</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Threshold</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ThresholdOtsuButton">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>430</y>
      <width>131</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Threshold (otsu)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="DilateButton">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>390</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Dilate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="GradientButton">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>390</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Gradient</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ErodeButton">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>430</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Erode</string>
    </property>
   </widget>
   <widget class="QPushButton" name="TopHatButton">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>430</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>TopHat</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OpenButton">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>470</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Open</string>
    </property>
   </widget>
   <widget class="QPushButton" name="BlackHatButton">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>470</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>BlackHat</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ClosingButton">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>510</y>
      <width>99</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Closing</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>560</y>
      <width>101</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>ExecuteTime:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>390</y>
      <width>41</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="msTime">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>560</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1496</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
